internal/client/errors.go

```diff
@@ -0,0 +1,46 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+type apiResponseError struct {
+	body   string
+	status uint16
+}
+
+func newAPIError(resp *http.Response) apiResponseError {
+	defer resp.Body.Close()
+	respBody, err := io.ReadAll(resp.Body)
+
+	status := uint16(resp.StatusCode)
+	body := ""
+	if err != nil {
+		body = fmt.Sprintf("failed to read response body. reason: %s", err.Error())
+	} else {
+		body = string(respBody)
+	}
+
+	return apiResponseError{
+		body:   body,
+		status: uint16(status),
+	}
+}
+
+// implement errors.Error interface
+func (e apiResponseError) Error() string {
+	return fmt.Sprintf("%d: %s", e.status, e.body)
+}
+
+// implement errors.Is interface
+func (e apiResponseError) Is(target error) bool {
+	err, ok := target.(apiResponseError)
+	return ok && err.status == http.StatusNotFound
```

Comment:
The `Is` method should be for generic equality testing, e.g. `err.status == e.status`. Having it restrict the compare to `StatusNotFound` seems more like a concern for the `IsNotFound` method below.